cmake/modules/AddSwift.cmake

```diff
@@ -534,7 +534,7 @@ function(_add_swift_runtime_link_flags target relpath_to_lib_dir bootstrapping)
       # installed host toolchain.
       get_filename_component(swift_bin_dir ${SWIFT_EXEC_FOR_SWIFT_MODULES} DIRECTORY)
       get_filename_component(swift_dir ${swift_bin_dir} DIRECTORY)
-      set(host_lib_dir "${swift_dir}/lib/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
+      set(host_lib_dir "${swift_dir}/lib/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}/${SWIFT_HOST_VARIANT_ARCH}")
```

Comment:
Does this preserve compatibility with existing toolchains?

Reply:
Good point, no, it will not on these ELF platforms. I always build with the latest Swift compiler though, so if others do the same, it will work fine. There may be a way to keep compatibility, but as I noted above about continuing to keep these libraries in the module directory when bootstrapping, I am not about to dig through the CMake config to add that for a usecase that almost nobody is likely using (I currently have bootstrapping turned off on Android because of #60272).

Reply:
> I currently have bootstrapping turned off on Android because of #60272

Out of curiosity, won’t this make Regex literal (which should be an official Swift 5.7 feature) unsupported? I suppose (at least the Swift piece of) bootstrapping is mandatory now.

Reply:
Yes, regex literals don't work for the upcoming Swift 5.8 that I have been testing out natively on Android, buttaface/termux-packages@a1befe6, but they should work on the current 5.7.3 release with bootstrapping enabled, as 5.7 was branched before those LLVM issues discussed in that bug.